driver-core/src/test/java/com/datastax/driver/core/querybuilder/QueryBuilderTest.java

```diff
@@ -300,6 +300,24 @@ public void updateTest() throws Exception {
     }
 
     @Test(groups = "unit")
+    public void should_handle_update_if_exists() {
+        Statement update = update("foo").with(set("x", 3)).ifExists();
+        assertThat(update.toString()).isEqualTo("UPDATE foo SET x=3 IF EXISTS;");
+    }
+    
+    @Test(groups = "unit")
+    public void should_handle_update_with_conditions_and_if_exists() {
+        Statement update = update("foo").with(set("x", 3)).where(eq("k", 2)).ifExists();
+        assertThat(update.toString()).isEqualTo("UPDATE foo SET x=3 WHERE k=2 IF EXISTS;");
+    }
+    
+    @Test(groups = "unit")
+    public void if_exists_should_combine_with_any_condition() {
```

Comment:
Yes. please remove these methods; the remaining one is IMHO the only one that should be added to the API.